### ISSUE TITLE
[Bug]: A11y Filtering Should be AND

### ISSUE THREAD

**Issue by ColinBuyck** [bloom-housing/bloom#4885]
### What is the issue?

When testing the a11y features filtering, it was noticed that multiple selections are beeing combined with an OR rather than an AND. As in, the results may only have one of the following required a11y features. This filter is a special case and should be built with an AND.

### How can you reproduce the bug?

When filtering is merged, it can be seen there.

### Is there any additional context?

Backend Change

### What browsers are you seeing the problem on?

Chrome

### Where are you seeing this issue?

Desktop

**Comment by skeller-exy** [comment]
LGTM 🥳
